gorgone/gorgone/modules/core/proxy/class.pm

```diff
@@ -494,18 +498,23 @@ sub periodic_exec {
     foreach (keys %{$connector->{clients}}) {
         if (defined($connector->{clients}->{$_}->{delete}) && $connector->{clients}->{$_}->{delete} == 1) {
             $connector->send_internal_action({
-                action => 'PONGRESET',
-                data => '{ "data": { "id": ' . $_ . ' } }',
+                action        => 'PONGRESET',
+                data          => '{ "data": { "id": ' . $_ . ' } }',
                 data_noencode => 1,
-                token => $connector->generate_token(),
-                target => ''
+                token         => $connector->generate_token(),
+                target        => ''
             });
             if (defined($connector->{clients}->{$_}->{class})) {
-	    	$connector->{clients}->{$_}->{class}->close();
-		$connector->{clients}->{$_}->{class}->cleanup();
-	    }
-            $connector->{clients}->{$_}->{class} = undef;
-            $connector->{clients}->{$_}->{delete} = 0;
+                $connector->{clients}->{$_}->{class}->close();
+                $connector->{clients}->{$_}->{class}->cleanup();
+            }
+            # if the connection to the node is not established, we stop listenning for new event for this destination,
+            # so event will be stored in zmq buffer until we start processng them again (see proxy_addnode)
+            # zmq queue have a limit in size (high water mark), so if the node never connect we will loose some message,
+            # stoping us from memory leak or other nasty problem.
```

Comment:
zmq queues have a size limit (high water mark), so if the node never connects, we lose some messages, preventing us from having memory leaks or other inconvenient problems.